### DISCUSSION TITLE
Only change array entries that match predicate

### DISCUSSION THREAD

**Discussion by sschneider-ihre-pvs** [selfrefactor/rambda#677] [Q&A]
Is there a way where I have a filter function that looks for a certain thing and only edit the filtered entries but keeping also the unchanged entries?
example:
```js
[1,2,3,'a', 4] -> filter(x => typeof x === 'string) -> map(x => parseInt(x, 16)) -> [1,2,3,10,4]
```
what I am trying to avoid is something like

```js
[1,2,3,'a', 4] -> map(x => typeof x === 'number' ? x : parseInt(x, 16)) -> [1,2,3,10,4]
```

**Comment by sschneider-ihre-pvs** [comment]
I was thinking of something like fork/tee or something. because if I filter, I will loose the information about at what index is the value that has been altered so I cannot merge with the original array to overwrite the value.

**Comment by selfrefactor** [comment]
This case is more about `Maybe` concept. FP-TS offers such methods. I think I already mentioned it in relation to `R.either` being used as Either Monad.

Unfortunately, my plans for Rambda are more about fixing existing issues and keeping up with Ramda. But I like your suggestion if you can find such function (as `fork`) being used elsewhere(other library or language). If you find such case, then I can include the method in `Rambdax`. Sorry for not being able to be more helpful than that.
